cmd/lakefs/cmd/run.go

```diff
@@ -113,8 +107,13 @@ var runCmd = &cobra.Command{
 				Error("Block adapter NOT SUPPORTED for production use")
 		}
 		metadata := stats.NewMetadata(ctx, logger, blockstoreType, authMetadataManager, cloudMetadataProvider)
-		bufferedCollector := stats.NewBufferedCollector(metadata.InstallationID, blockStore.RuntimeStats, cfg)
-
+		bufferedCollector := stats.NewBufferedCollector(metadata.InstallationID, cfg)
+		// init block store
+		blockStore, err := factory.BuildBlockAdapter(context.WithValue(ctx, block.ContextKeyStatsCollector, bufferedCollector), cfg)
```

Comment:
Talked f2f over passing the stats collector as a parameter to the factory.
The factory will pass the collector using the specific adapter options.
The adapter can define the interface of how to send stats and not directly use the stats package.